app/models/petition.rb

```diff
@@ -793,6 +793,36 @@ def validate_creator!(now = Time.current)
     end
   end
 
+  def increment_signature_count!(time = Time.current)
+   sql = "signature_count = signature_count + ?, last_signed_at = ?, updated_at = ?"
+   count = signatures.validated_count(last_signed_at, time)
+   return false if count.zero?
+   if result = update_all([sql, count, time, time]) > 0
+     self.reload
+     updates = []
+     if at_threshold_for_moderation? && collecting_sponsors?
+       updates << "state = '#{SPONSORED_STATE}'"
+       updates << "moderation_threshold_reached_at = :now"
+     elsif pending?
+       updates << "state = '#{VALIDATED_STATE}'"
+     end
+     if at_threshold_for_referral?
+       updates << "referral_threshold_reached_at = :now"
+     end
+     if at_threshold_for_debate?
+       updates << "debate_threshold_reached_at = :now"
+       updates << "debate_state = 'awaiting'"
+     end
+     updates = updates.join(", ")
+     if updates.present?
+       if update_all([updates, now: time]) > 0
+         self.reload
+       end
+     end
+   end
+   result
+ end
+
```

Comment:
Sorry, this method already exists on the Petition model. I was showing you how it worked - sorry for the confusion.